pkg/k8s/apis/cilium.io/v2/register.go

```diff
@@ -171,7 +231,9 @@ func createUpdateCRD(clientset apiextensionsclient.Interface, CRDName string, cr
 			}
 
 			if clusterCRD.Spec.Validation == nil ||
-				!reflect.DeepEqual(clusterCRD.Spec.Validation.OpenAPIV3Schema, crv.OpenAPIV3Schema) {
+				crd.Spec.Validation == nil ||
```

Comment:
If you have that comment in line 195, this will always be true

Reply:
I finally looked at this. It is true but we only get here if the version has changed and needsUpdate returns true. At that point we do want to update it if either is nil in case we add or remove validation at some point.

Reply:
Please rebase with master, I've changed this

Reply:
done